### PR TITLE
Added Redis Cluster cookbook

### PR DESCRIPTION
**Pull request by RajeshwariMakond**
Please Review the PR

### CODE REVIEW

**Review by aashishmodak — COMMENTED** [review]
- fix master and slave cookbook or idempotency. When the service is stopped, it should restart it.
- password should go in databag, not in attribute file
- master ip and port should be configurable while executing cookbook (check how to pass attribute to cookbook)

File: redis-master/attributes/default/default.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+default['redis-master']['tcp-value'] = "60"
+default['redis-master']['mpsa-value'] = Chef::EncryptedDataBagItem.load('Redis', 'redis', 'redis')['password']
```
Comment (aashishmodak) [inline redis-master/attributes/default/default.rb:2]: are you using this attribute? Dont see it is getting used

File: redis-master/recipes/master.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+
+password = Chef::EncryptedDataBagItem.load('Redis', 'redis', 'redis')['password']
```
Comment (aashishmodak) [inline redis-master/recipes/master.rb]: are we using this anywhere?

File: redis-master/recipes/slave.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
+passwordslave = Chef::EncryptedDataBagItem.load('Redis', 'redis', 'redis')['passwordslave']
```
Comment (aashishmodak) [inline redis-master/recipes/slave.rb]: are we using this anywhere?

File: redis-master/attributes/default/default.rb (hunk 1 above)
Comment (RajeshwariMakond) [reply redis-master/attributes/default/default.rb:2]: Yes, Used all the attributes mentioned in attributes/default.rb file

File: redis-master/recipes/slave.rb (hunk 3 above)
Comment (RajeshwariMakond) [reply redis-master/recipes/slave.rb]: Actually, while verification we need to use that.

Comment (aashishmodak) [reply redis-master/recipes/slave.rb]: lets remove it from cookbook

**Review by aashishmodak — APPROVED** [review]

**Pull request merged**